pkgs/sdk_triage_bot/lib/src/gemini.dart

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:google_generative_ai/google_generative_ai.dart';
+import 'package:http/http.dart' as http;
+
+abstract class GeminiService {
+  Future<String> summarize(String prompt);
+  Future<List<String>> classify(String prompt);
+}
+
+class GeminiServiceImpl implements GeminiService {
```

Comment:
[nit] I don't think we're gaining any utility from splitting into an interface and `Impl` - can we combine these for now?

Reply:
This separation help make clear what the mocks should implement / the boundaries of the interface. Were you thinking of something like having the mocks just implement the implementations directly?

Reply:
Yes - having test doubles use the interface from the implementations is the common pattern. The impl should be able to keep any non-interface members private, so from the outside the interface and test doubles won't be any different.

Thanks to the Dart paradigm of all-classes-are-interfaces (by default at least)`*Impl` classes are typically only used in Dart for cases where the impl needs heavyweight dependencies that a modular build can elide when only the interface is required.

Reply:
OK - I did remove the extra interfaces. The code is a bit simpler (and I was able to hide the non public interface portions of the implementation classes w/ private fields).